### PR TITLE
Add a new parameter that allows padding in the buffers used for IO.

### PR DESCRIPTION
**Pull request by mattkur**
This is to exercise certain code paths where the IO coming from adjacent memory pages affects IO performance.

### CODE REVIEW

File: CmdLineParser/CmdLineParser.cpp [hunk 1]
```diff
@@ -173,6 +173,7 @@ void CmdLineParser::_DisplayUsageInfo(const char *pszFilename) const
     printf("  -i<count>             number of IOs per burst; see -j [default: inactive]\n");
     printf("  -j<milliseconds>      interval in <milliseconds> between issuing IO bursts; see -i [default: inactive]\n");
     printf("  -I<priority>          Set IO priority to <priority>. Available values are: 1-very low, 2-low, 3-normal (default)\n");
+    printf("  -k<align>[K|M|G|b]    Align the buffers for each IO packet on this bounday\n");
```
Comment (dl2n) [inline CmdLineParser/CmdLineParser.cpp]: boundary

File: CmdLineParser/CmdLineParser.cpp [hunk 2]
```diff
@@ -822,6 +823,29 @@ bool CmdLineParser::_ReadParametersFromCmdLine(const int argc, const char *argv[
...
+                if (_GetSizeInBytes(arg + 1, cb))
+                {
+                    for (auto i = vTargets.begin(); i != vTargets.end(); i++)
+                    {
+                        i->SetIOBufferAlignment(cb);
+                    }
+                }
+                else
+                {
+                    fprintf(stderr, "Invalid IO buffer alignmentpassed to -k\n");
```
Comment (dl2n) [inline CmdLineParser/CmdLineParser.cpp]: alignment passed

**Comment by dl2n** [comment]
Is this actually guaranteeing alignment of the buffers v. separation of the IO? I think alignment would require some more work to handle the VA alignment of whatever VirtualAlloc returned.

**Comment by mattkur** [comment]
No, it doesn't actually. I didn't do the work to guarantee that the VA offset is correct. Alignment is probably a poor choice of words. I'm okay holding off on accepting this change until we can guarantee the alignment.

File: IORequestGenerator/IORequestGenerator.cpp [hunk 3]
```diff
@@ -1013,7 +1013,7 @@ DWORD WINAPI threadFunc(LPVOID cookie)
     // adjust thread token if large pages are needed
     for (auto pTarget = p->vTargets.begin(); pTarget != p->vTargets.end(); pTarget++)
     {
-        if (pTarget->GetUseLargePages())
+        if (pTarget->GetUseLargePages() | pTarget->GetIOBufferAlignmentSpecified())
```
Comment (DanPear) [inline IORequestGenerator/IORequestGenerator.cpp]: Shouldn't this be defined as || and not |?

Comment (dl2n) [reply IORequestGenerator/IORequestGenerator.cpp]: Both would work but yes, I'd agree as a matter of practice that using the Boolean test is preferable.

**Pull request closed without merging**